proxmox-image fails to build with qemu 7.0

### Describe the bug
After qemu: 6.2.0 -> 7.0.0 (commit 9d733f3d4975fb3c516a2bf2bf1bc6f50781a553) proxmox-image fails to build.

### Steps To Reproduce
Steps to reproduce the behavior:
`nix run github:nix-community/nixos-generators -- -f proxmox -I nixpkgs=https://releases.nixos.org/nixos/unstable/nixos-22.05pre375061.c777cdf5c56/nixexprs.tar.xz`

### Expected behavior
Command should produce a vma image

### Actual behavior
Various compilation errors (copied in case others are searching for a fix)
FAILED: libblock.fa.p/vma-writer.c.o
 ../vma-writer.c:206:5: error: too few arguments to function 'aio_set_fd_handler'
../vma-writer.c:209:5: error: too few arguments to function 'aio_set_fd_handler'

### Additional context
We probably need a rebased patch from proxmox:
https://git.proxmox.com/?p=pve-qemu.git;a=history;f=debian/patches/pve/0026-PVE-Backup-add-vma-backup-format-code.patch;h=b37b17c286da3d32945fbee8ee4fd97a418a50db;hb=HEAD

### Notify maintainers
@illustris @SuperSandro2000

--- a/nixos/modules/virtualisation/proxmox-image.nix
+++ b/nixos/modules/virtualisation/proxmox-image.nix
@@ -127,16 +127,26 @@ with lib;
       name = "proxmox-${cfg.filenameSuffix}";
       postVM = let
         # Build qemu with PVE's patch that adds support for the VMA format
-        vma = pkgs.qemu_kvm.overrideAttrs ( super: {
+        vma = pkgs.qemu_kvm.overrideAttrs ( super: rec {
+
+          # proxmox's VMA patch doesn't work with qemu 7.0 yet
+          version = "6.2.0";
+          src = pkgs.fetchurl {
+            url= "https://download.qemu.org/qemu-${version}.tar.xz";
+            hash = "sha256-aOFdjkWsVjJuC5pK+otJo9/oq6NIgiHQmMhGmLymW0U=";
+          };
+
           patches = let
-            rev = "cc707c362ea5c8d832aac270d1ffa7ac66a8908f";
-            path = "debian/patches/pve/0025-PVE-Backup-add-vma-backup-format-code.patch";
+            rev = "b37b17c286da3d32945fbee8ee4fd97a418a50db";
+            path = "debian/patches/pve/0026-PVE-Backup-add-vma-backup-format-code.patch";
             vma-patch = pkgs.fetchpatch {
-              url = "https://git.proxmox.com/?p=pve-qemu.git;a=blob_plain;hb=${rev};f=${path}";
-              sha256 = "1z467xnmfmry3pjy7p34psd5xdil9x0apnbvfz8qbj0bf9fgc8zf";
+              url = "https://git.proxmox.com/?p=pve-qemu.git;a=blob_plain;h=${rev};f=${path}";
+              hash = "sha256-siuDWDUnM9Zq0/L2Faww3ELAOUHhVIHu5RAQn6L4Atc=";
             };
-          in super.patches ++ [ vma-patch ];
+          in [ vma-patch ];
+
           buildInputs = super.buildInputs ++ [ pkgs.libuuid ];
+
         });
       in
       ''

--- a/nixos/release.nix
+++ b/nixos/release.nix
@@ -221,6 +221,29 @@ in rec {
 
   );
 
+  # KVM image for proxmox in VMA format
+  proxmoxImage = forMatchingSystems [ "x86_64-linux" ] (system:
+    with import ./.. { inherit system; };
+
+    hydraJob ((import lib/eval-config.nix {
+      inherit system;
+      modules = [
+        ./modules/virtualisation/proxmox-image.nix
+      ];
+    }).config.system.build.VMA)
+  );
+
+  # LXC tarball for proxmox
+  proxmoxLXC = forMatchingSystems [ "x86_64-linux" ] (system:
+    with import ./.. { inherit system; };
+
+    hydraJob ((import lib/eval-config.nix {
+      inherit system;
+      modules = [
+        ./modules/virtualisation/proxmox-lxc.nix
+      ];
+    }).config.system.build.tarball)
+  );
 
   # A disk image that can be imported to Amazon EC2 and registered as an AMI
   amazonImage = forMatchingSystems [ "x86_64-linux" "aarch64-linux" ] (system:

